src/app/geo/init-map.directive.js

```diff
@@ -99,6 +99,15 @@ function rvInitMap($rootScope, configService, geoService, events, referenceServi
                 return $compile(html)($rootScope);
             };
 
+            // reference to the angular compiler that also returns the scope
+            apiMap.$compileWithScope = function(html) {
+                const scope = $rootScope.$new(true)
+                return {
+                    html: $compile(html)(scope),
```

Comment:
Does the html need to be returned here? It was provided to the function so the caller should have a reference to it.

Reply:
I thought `$compile` returns the new html after the input's compiled through angular so we would need the html returned there.

Reply:
> Calling the linking function returns the element of the template. It is either the original element passed in, or the clone of the element if the cloneAttachFn is provided.
https://docs.angularjs.org/api/ng/service/$compile

Since `cloneAttachFn` is not passed, the original element passed in is altered.